openupgrade_scripts/scripts/payment/16.0.2.0/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,417 @@
+# NOTHING TO DO
+
+NEW ir.ui.view: payment.payment_provider_form
+# NOTHING TO DO: already done, see comment below
+
+NEW ir.ui.view: payment.payment_provider_kanban
+# NOTHING TO DO: already done, see comment below
+
+NEW ir.ui.view: payment.payment_provider_list
+# NOTHING TO DO: already done, see comment below
+
+NEW ir.ui.view: payment.payment_provider_onboarding_wizard_form
+# NOTHING TO DO: already done, see comment below
+
+NEW ir.ui.view: payment.payment_provider_search
+# NOTHING TO DO: already done, see comment below
+
+DEL ir.ui.view: payment.account_invoice_view_form_inherit_payment
+# DONE: pre-migration: moved to account_payment
+
+DEL ir.ui.view: payment.onboarding_payment_acquirer_step
+# NOTHING TO DO: it seams that the feature has been moved to other modules
+
+DEL ir.ui.view: payment.payment_acquirer_form
+# DONE: pre-migration: renamed payment_acquirer_form -> payment_provider_form
+
+DEL ir.ui.view: payment.payment_acquirer_kanban
+# DONE: pre-migration: renamed payment_acquirer_kanban -> payment_provider_kanban
+
+DEL ir.ui.view: payment.payment_acquirer_list
+# DONE: pre-migration: renamed payment_acquirer_list -> payment_provider_list
+
+DEL ir.ui.view: payment.payment_acquirer_onboarding_wizard_form
+# DONE: pre-migration: renamed payment_acquirer_onboarding_wizard_form -> payment_provider_onboarding_wizard_form
+
+DEL ir.ui.view: payment.payment_acquirer_search
+# DONE: pre-migration: renamed payment_acquirer_search -> payment_provider_search
+
+DEL ir.ui.view: payment.payment_refund_wizard_view_form
+# DONE: pre-migration: moved to account_payment
+
+DEL ir.ui.view: payment.view_account_journal_form
+# DONE: pre-migration: moved to account_payment
+
+DEL ir.ui.view: payment.view_account_payment_form_inherit_payment
+# DONE: pre-migration: moved to account_payment
+
+DEL ir.ui.view: payment.view_account_payment_register_form_inherit_payment
+# DONE: pre-migration: moved to account_payment
+
+DEL payment.acquirer: payment.payment_acquirer_adyen (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_alipay (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_authorize (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_buckaroo (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_mollie (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_ogone (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_paypal (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_payulatam (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_payumoney (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_sepa_direct_debit (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_sips (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_stripe (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_test (noupdate)
+DEL payment.acquirer: payment.payment_acquirer_transfer (noupdate)
+# NOTHING TO DO: replaced by payment.provider element (see below)
```

Comment:
Are we sure about this? The user could have put modifications in these records that will not be carried over to the newly created records.

Reply:
I'm pretty sure there is no change by the user on theses data. It contains data like this:

```xml

    <record id="payment_acquirer_adyen" model="payment.acquirer">
        <field name="name">Adyen</field>
        <field name="display_as">Credit Card (powered by Adyen)</field>
        <field name="image_128" type="base64" file="payment_adyen/static/src/img/adyen_icon.png"/>
        <field name="module_id" ref="base.module_payment_adyen"/>
        <field name="description" type="html">
            <p>
                A payment gateway to accept online payments via credit cards, debit cards and bank
                transfers.
            </p>
            <ul class="list-inline">
                <li class="list-inline-item"><i class="fa fa-check"/>Online Payment</li>
                <li class="list-inline-item"><i class="fa fa-check"/>Payment Status Tracking</li>
            </ul>
        </field>
        <!-- https://www.adyen.com/payment-methods -->
        <field name="payment_icon_ids"
               eval="[(6, 0, [
                   ref('payment.payment_icon_cc_bancontact'),
                   ref('payment.payment_icon_cc_maestro'),
                   ref('payment.payment_icon_cc_mastercard'),
                   ref('payment.payment_icon_cc_visa'),
                   ref('payment.payment_icon_cc_discover'),
                   ref('payment.payment_icon_cc_diners_club_intl'),
                   ref('payment.payment_icon_cc_jcb'),
                   ref('payment.payment_icon_cc_unionpay'),
               ])]"/>
    </record>
```

to 

```xml
    <record id="payment_provider_adyen" model="payment.provider">
        <field name="name">Adyen</field>
        <field name="display_as">Credit Card (powered by Adyen)</field>
        <field name="image_128" type="base64" file="payment_adyen/static/description/icon.png"/>
        <field name="module_id" ref="base.module_payment_adyen"/>
        <!-- https://www.adyen.com/payment-methods -->
        <field name="payment_icon_ids"
               eval="[(6, 0, [
                   ref('payment.payment_icon_cc_bancontact'),
                   ref('payment.payment_icon_cc_maestro'),
                   ref('payment.payment_icon_cc_mastercard'),
                   ref('payment.payment_icon_cc_visa'),
                   ref('payment.payment_icon_cc_discover'),
                   ref('payment.payment_icon_cc_diners_club_intl'),
                   ref('payment.payment_icon_cc_jcb'),
                   ref('payment.payment_icon_cc_unionpay'),
               ])]"/>
    </record>
```

Reply:
Users can absolutely modify these records.

![Screenshot from 2023-07-03 18-09-56](https://github.com/OCA/OpenUpgrade/assets/12065945/7b7159a4-3bce-41eb-bdb4-fe167322d02a)

In any case, currently after migration, you just get duplicates of everything:

```sql
> select id, name from payment_provider where cast(name->'en_US' as text) ilike '%mollie%';
+----+---------------------+
| id | name                |
|----+---------------------|
| 5  | {"en_US": "Mollie"} |
| 23 | {"en_US": "Mollie"} |
+----+---------------------+
SELECT 2
```

Reply:
Hum, indeed. I did not watch other field of the model. I will look deeper at this.

Reply:
I think that I fixed it. :)

Renamed xmlids, added payment_provider in noupdate_changes, and loaded it.